Client/Frontend/Browser/Tab.swift

```diff
@@ -146,6 +146,45 @@ class Tab: NSObject {
         }
         return self.url
     }
+    
+    var isFxHomeTab: Bool {
+        if let numberOfUrls = self.sessionData?.urls.count,
+           let offset = self.sessionData?.currentPage,
+           let url = self.sessionData?.urls[numberOfUrls - 1 + offset],
+           url.absoluteString.hasPrefix("internal://") {
+            return true
+        }
+        
+        return false
+    }
+    
+    var isCustomHomeSet: Bool {
+        guard let profile = browserViewController?.profile else { return false }
+        
+        let isHomeSet = HomeButtonHomePageAccessors.getHomePage(profile.prefs)
+        if isHomeSet == nil {
+            return false
+        }
+        
+        return true
+    }
+    
+    var isCustomHomeTab: Bool {
+        guard let profile = self.browserViewController?.profile else { return false }
+        
+        if let customHomeUrl = HomeButtonHomePageAccessors.getHomePage(profile.prefs),
+           let numberOfUrls = self.sessionData?.urls.count,
```

Comment:
It might make more sense to have one variable `isAtHome` that does the appropriate check, versus having both and doing the check twice based on the users pref.